### PR TITLE
Create rpm and deb packages

### PR DESCRIPTION
**Pull request by matt-richardson**
Adds the creation of basic `rpm` and `deb` packages to the build process.

future work:
* add `octo` to the path
* figure out dependencies
* add the publish to the deploy process

### CODE REVIEW

File: BuildAssets/create-linux-packages.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# Remove existing packages, fpm doesnt like to overwrite
+rm *.{deb,rpm}
```
Comment (matt-richardson) [inline BuildAssets/create-linux-packages.sh]: This logs a nasty error to stderr if the files dont exist. Be nice if we could suppress that.

File: build.cake [hunk 2]
```diff
@@ -3,13 +3,17 @@
 //////////////////////////////////////////////////////////////////////
 #tool "nuget:?package=GitVersion.CommandLine&version=4.0.0"
 #tool "nuget:?package=ILRepack&version=2.0.13"
-#addin "nuget:?package=SharpCompress&version=0.12.4"
+#addin "nuget:?package=SharpCompress&version=0.24.0"
```
Comment (matt-richardson) [inline build.cake]: Needed to bump to get tar compressio

File: build.cake [hunk 3]
```diff
@@ -325,6 +337,45 @@ Task("BuildDockerImage")
...
+    DeleteDirectory(artifactsDir + $"/OctopusTools.{nugetVersion}.linux-x64.extracted", new DeleteDirectorySettings { Recursive = true, Force = true });
+ 
+    CreateDirectory($"{artifactsDir}/linuxpackages");
+    MoveFiles(GetFiles($"{artifactsDir}/*.deb"), $"{artifactsDir}/linuxpackages");
+    MoveFiles(GetFiles($"{artifactsDir}/*.rpm"), $"{artifactsDir}/linuxpackages");
+    TarGzip($"{artifactsDir}/linuxpackages", $"{artifactsDir}/OctopusTools.{nugetVersion}.linux-x64.packages");
+    var buildSystem = BuildSystemAliases.BuildSystem(Context);
+    buildSystem.TeamCity.PublishArtifacts($"{artifactsDir}/OctopusTools.{nugetVersion}.linux-x64.packages.tar.gz");
+    DeleteDirectory($"{artifactsDir}/linuxpackages", new DeleteDirectorySettings { Recursive = true, Force = true });
+
```
Comment (matt-richardson) [inline build.cake]:
```suggestion
```

File: BuildAssets/create-linux-packages.sh (hunk 1 above)
Comment (thedewi) [reply BuildAssets/create-linux-packages.sh]: The only way I know is
```suggestion
rm -f *.{deb,rpm}
```

File: build.cake (hunk 2 above)
Comment (thedewi) [reply build.cake]: 👍 Good, it was ancient

**Pull request merged**